status/flowcell.py

```diff
@@ -278,22 +278,29 @@ def fetch_barcodes(self, run_name):
             df.loc[df.bc_name == df.barcode_alias, "barcode_alias"] = ""
 
             # Calculate percentages
-            df["basecalled_pass_read_count_pc"] = np.where(
-                sum(df.basecalled_pass_read_count) > 0,
-                round(df.basecalled_pass_read_count / sum(df.basecalled_pass_read_count) * 100, 2),
-                0)
-            df["basecalled_pass_bases_pc"] = np.where(
-                sum(df.basecalled_pass_bases) > 0,
-                round(df.basecalled_pass_bases / sum(df.basecalled_pass_bases) * 100, 2),
-                0)
-            df["average_read_length_passed"] =  np.where(
-                df.basecalled_pass_read_count > 0,
-                round(df.basecalled_pass_bases / df.basecalled_pass_read_count, 2).astype(int),
-                0)
-            df["accuracy"] =  np.where(
-                df.basecalled_pass_bases + df.basecalled_fail_bases > 0,
-                round(df.basecalled_pass_bases / (df.basecalled_pass_bases + df.basecalled_fail_bases) * 100, 2),
-                0)
+            if sum(df.basecalled_pass_read_count) > 0:
```

Comment:
I'm not sure  but to me it would seem we need an 'else' clause here? We still use the "basecalled_pass_read_count_pc" column further down, and that would error if it wasn't defined?

Reply:
True, that would only happen if a run has no passed reads whatsoever, which hasn't happened yet.

Fix in [fc7c816](https://github.com/SciLifeLab/genomics-status/pull/721/commits/fc7c816dd32fcdb7c31436e71ab310c010fb6d7a)